robot_skills/src/robot_skills/arms.py

```diff
@@ -696,15 +749,19 @@ def _send_joint_trajectory(self, joints_references, timeout=rospy.Duration(5), j
             else:
                 joint_names = self.joint_names
 
-        time_from_start = rospy.Duration()
         ps = []
+        time_from_start = 0.0
+        start_joint_state = self.get_joint_states()
+        prev_joint_ref = [start_joint_state[jn] for jn in joint_names]
         for joints_reference in joints_references:
+            max_diff = sorted([abs(prev - new) for prev, new in zip(prev_joint_ref, joints_reference)])[-1]
             if len(joints_reference) != len(joint_names):
                 rospy.logwarn('Please use the correct %d number of joint references (current = %d'
                               % (len(joint_names), len(joints_references)))
+            time_from_start += max_diff/max_joint_vel
             ps.append(JointTrajectoryPoint(
                 positions=joints_reference,
-                time_from_start=time_from_start))
+                time_from_start=rospy.Duration.from_sec(time_from_start)))
```

Comment:
If I understand it correctly, the above code uses the max joint velocity along with relative position difference per joint to calculate `time_from_start`. Why not use ros time to track time instead of approximating it this way?

Reply:
`JointTrajectoryPoint` is a ROS message that has a `rospy.Duration` as the `time_from_start` field, assigning an `int` won't work

Reply:
> Why not use ros time to track time instead of approximating it this way?
Because we are not tracking time, we are computing the time!